moped-editor/src/views/projects/projectView/ProjectComponents/ControlledAutocomplete.js

```diff
@@ -0,0 +1,55 @@
+import { Autocomplete } from "@mui/material";
+import { Controller } from "react-hook-form";
+import { TextField } from "@mui/material";
+
+/**
+ * A react-hook-form wrapper of the MUI autocomplete component
+ * @return {JSX.Element}
+ */
+export default function ControlledAutocomplete({
+  id,
+  options,
+  renderOption,
+  name,
+  control,
+  label,
+  autoFocus = false,
+  multiple = false,
+  disabled,
+  helperText,
+  error,
+}) {
+  return (
+    <Controller
+      id={id}
+      name={name}
+      control={control}
+      render={({ field }) => (
+        <Autocomplete
+          {...field}
+          options={options}
+          multiple={multiple}
+          getOptionLabel={(option) => option?.label || ""}
+          isOptionEqualToValue={(option, value) =>
+            option?.value === value?.value
+          }
+          renderOption={renderOption}
+          disabled={disabled}
+          renderInput={(params) => (
+            <TextField
+              {...params}
+              inputRef={field.ref}
+              size="small"
+              label={label}
+              variant="outlined"
+              autoFocus={autoFocus}
+              helperText={helperText}
+              error={error}
```

Comment:
the `error` and `helperText` props are the only additions/changes to this code other than extracting it from the `utils/form.js` file.